cms/core/blocks/base.py

```diff
@@ -36,20 +44,45 @@ def get_link(self, context: dict | None = None) -> dict[str, str] | None:
             if has_description:
                 value["description"] = desc or getattr(page.specific_deferred, "summary", "")
 
+            content_type_label = get_content_type_for_page(page)
+            page_release_date = getattr(page.specific_deferred, "release_date", None)
+
+        if not value:
+            return None
+
+        if content_type := self.get("content_type"):
+            content_type_label = get_related_content_type_label(content_type)
+
+        release_date: datetime = page_release_date or self.get("release_date")
```

Comment:
```suggestion
        release_date: datetime = self.get("release_date") or page_release_date
```

that is, if manually added it takes precendence over the page-derived one